### PR TITLE
Fix  to pick pinned  version of  chef-foundation  on windows

### PR DESCRIPTION
**Pull request by poorndm**
## Description
On the Windows build agent, instead of the pinned chef-foundation version as defined in [.buildkite-platform.json](https://github.com/chef/chef/blob/main/.buildkite-platform.json#L2) file, the latest version of chef-foundation gets installed. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: .expeditor/scripts/omnibus_chef_build.ps1 [hunk 1]
```diff
@@ -28,7 +28,9 @@ $env:ARTIFACTORY_BASE_PATH="com/getchef"
 $env:ARTIFACTORY_ENDPOINT="https://artifactory-internal.ps.chef.co/artifactory"
 $env:ARTIFACTORY_USERNAME="buildkite"
 
-Write-Output "--- Install Chef Foundation"
+
+$CHEF_FOUNDATION_VERSION = $env:CHEF_FOUNDATION_VERSION
+Write-Output "--- Installing Chef Foundation $CHEF_FOUNDATION_VERSION"
 . { Invoke-WebRequest -useb https://omnitruck.chef.io/chef/install.ps1 } | Invoke-Expression; install -channel "current" -project "chef-foundation" -v $CHEF_FOUNDATION_VERSION
```
Comment (vkarve-chef) [inline .expeditor/scripts/omnibus_chef_build.ps1:34]: how about using the `$env:CHEF_FOUNDATION_VERSION` notation here directly? i.e.
`. { Invoke-WebRequest -useb https://omnitruck.chef.io/chef/install.ps1 } | Invoke-Expression; install -channel "current" -project "chef-foundation" -v $env:CHEF_FOUNDATION_VERSION`

File: .buildkite/build-test-omnibus.sh [hunk 2]
```diff
@@ -122,7 +122,7 @@ then
       echo "      - \"-Command\""
       echo "      propagate-environment: true"
       echo "      environment:"
-      echo "        - CHEF_FOUNDATION_VERSION"
+      echo "        - CHEF_FOUNDATION_VERSION=$CHEF_FOUNDATION_VERSION"
```
Comment (vkarve-chef) [inline .buildkite/build-test-omnibus.sh:125]: This may not have to change likely

File: .expeditor/scripts/omnibus_chef_build.ps1 (hunk 1 above)
Comment (poorndm) [reply .expeditor/scripts/omnibus_chef_build.ps1:34]: @vkarve-chef
I tried the above fix. It throws the below error.

At C:\workdir\.expeditor\scripts\omnibus_chef_build.ps1:31 char:45
--
  | + Write-Output "--- Install Chef Foundation : $env:${CHEF_FOUNDATION_VE ...
  | +                                             ~~~~~
  | Variable reference is not valid. ':' was not followed by a valid variable name
  | character. Consider using ${} to delimit the name.
  | At C:\workdir\.expeditor\scripts\omnibus_chef_build.ps1:32 char:152
  | + ... nstall -channel "current" -project "chef-foundation" -v $env:$CHEF_FO ...
  | +                                                             ~~~~~
  | Variable reference is not valid. ':' was not followed by a valid variable name
  | character. Consider using ${} to delimit the name.
  | + CategoryInfo          : ParserError: (:) [], ParseException
  | + FullyQualifiedErrorId : InvalidVariableReferenceWithDrive
  ========
  
  Trying with adding  ${}

Comment (poorndm) [reply .expeditor/scripts/omnibus_chef_build.ps1]: . { Invoke-WebRequest -useb https://omnitruck.chef.io/chef/install.ps1 } | Invoke-Expression; install -channel "current" -project "chef-foundation" -**v ${env:CHEF_FOUNDATION_VERSION}** - went well with this changes

File: .buildkite/build-test-omnibus.sh (hunk 2 above)
Comment (sean-simmons-progress) [reply .buildkite/build-test-omnibus.sh]: It should not have to change, and its required for setting an ENV var and passing it to docker.

File: .expeditor/scripts/omnibus_chef_build.ps1 (hunk 1 above)
Comment (sean-simmons-progress) [reply .expeditor/scripts/omnibus_chef_build.ps1]: the change looks good to me and simplifies binding the variable.

**Review by jaymzh — CHANGES_REQUESTED** [review]

**Comment by tpowell-progress** [comment]
@poorndm Could we fill out the description on this PR for posterity, please?

File: .expeditor/scripts/omnibus_chef_build.ps1 [hunk 3]
```diff
@@ -28,8 +28,9 @@ $env:ARTIFACTORY_BASE_PATH="com/getchef"
 $env:ARTIFACTORY_ENDPOINT="https://artifactory-internal.ps.chef.co/artifactory"
 $env:ARTIFACTORY_USERNAME="buildkite"
 
-Write-Output "--- Install Chef Foundation"
-. { Invoke-WebRequest -useb https://omnitruck.chef.io/chef/install.ps1 } | Invoke-Expression; install -channel "current" -project "chef-foundation" -v $CHEF_FOUNDATION_VERSION
+
+Write-Output "--- Installing Chef Foundation $CHEF_FOUNDATION_VERSION"
```
[the remark this answers is not in the document]
Comment (sean-simmons-progress) [reply .expeditor/scripts/omnibus_chef_build.ps1:32]: yes, it should.

**Review by jaymzh — APPROVED** [review]
Approved, assuming tests pass.

**Review by johnmccrae — APPROVED** [review]

**Comment by jaymzh** [comment]
SonarQube known to fail. Merging.

**Pull request merged**